src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -768,107 +761,187 @@ void RNN::fillSequenceDesc() {
     }
 
     VectorDims dims_b = { L, D, Gb, SC };
 
-    auto _data_type = DnnlExtensionUtils::ElementTypeToDataType(Prec);
-    auto w_bias_data_desc = DnnlBlockedMemoryDesc(Shape(dims_b), _data_type, getWeightsFormatTagByDims(dims_b));
-    MemoryPtr w_bias_data_mem = std::make_shared<Memory>(getEngine(), w_bias_data_desc);
-    auto b_ptr = static_cast<dataType*>(w_bias_data_mem->getData());
-    if (b_ptr == nullptr)
-        THROW_CPU_NODE_ERR("has unallocated internal blob.");
-
-    auto *constInputNode = dynamic_cast<Input *>(getParentEdgeAt(bIdx)->getParent().get());
-    auto constBlob = constInputNode->getMemoryPtr();
-    auto const elementsCount = constBlob->getSize() / constBlob->getDesc().getPrecision().size();
-
-    std::vector<dataType> ie_b_vec(elementsCount);
-    cpu_convert(constBlob->getData(),
-                &ie_b_vec[0],
-                DnnlExtensionUtils::DataTypeToElementType(constBlob->getDataType()),
-                Prec,
-                elementsCount);
-
-    for (size_t g = 0; g < Gb; g++) {
-        dataType *l_b_ptr = b_ptr + gate_map[g] * SC;
-        const dataType *l_ie_b_ptr = &ie_b_vec[g * SC];
-        cpu_memcpy(l_b_ptr, l_ie_b_ptr, SC * sizeof(typename element_type_traits<Prec>::value_type));
+    auto dnnl_type = DnnlExtensionUtils::ElementTypeToDataType(ET);
+    auto w_bias_data_desc = std::make_shared<DnnlBlockedMemoryDesc>(Shape(dims_b), dnnl_type, getWeightsFormatTagByDims(dims_b));
+
+    auto create = [&]() {
+        MemoryPtr w_bias_data_mem = std::make_shared<Memory>(getEngine(), w_bias_data_desc);
+        auto b_ptr = reinterpret_cast<DataType*>(w_bias_data_mem->getData());
+        if (b_ptr == nullptr) {
+            THROW_CPU_NODE_ERR("has unallocated internal blob.");
+        }
+
+        std::vector<DataType> ie_b_vec;
+        DataType* ie_b_ptr = nullptr;
+
+        if (dnnl_type != b_const_blob->getDataType()) {
+            const size_t ie_b_vec_size = getInputShapeAtPort(bIdx).getElementsCount();
+            ie_b_vec.resize(ie_b_vec_size);
+            ie_b_ptr = ie_b_vec.data();
+
+            cpu_convert(b_const_blob->getData(), ie_b_ptr, DnnlExtensionUtils::DataTypeToElementType(b_const_blob->getDataType()), ET, ie_b_vec_size);
+        } else {
+            ie_b_ptr = reinterpret_cast<DataType*>(b_const_blob->getData());
+        }
+
+        const uint64_t step = SC * sizeof(DataType);
+        parallel_for(Gb, [&](size_t g) {
+            DataType* l_b_ptr = b_ptr + m_gate_map[g] * SC;
+            const DataType* l_ie_b_ptr = ie_b_ptr + g * SC;
+            cpu_memcpy(l_b_ptr, l_ie_b_ptr, step);
+        });
+
+        return w_bias_data_mem;
+    };
+
+    if (auto weight_cache = context->getWeightsCache()) {
+        const std::string hash_str = getName() + "_2_" + std::to_string(dnnl::impl::primitive_hashing::get_md_hash(*w_bias_data_desc->getDnnlDesc().get()));
```

Comment:
Theoreticall node Name might not be unique. Why cannot we use pointer on b_const_blob data instead?

Reply:
Hash Pointer+Primitive does not work for this node. Pointer is changed after reorder. On update we have a new Primitive only, so we cannot combine such hash.